server.go

```diff
@@ -192,62 +192,85 @@ func (s *Server) Close() {
 		s.ExitProcess()
 	}
 
-	s.statusLogger.Info("closing tcp listener in %s", s.closeListenerDelay.String())
+	s.Logger.Info("closing tcp listener in %s", s.closeListenerDelay.String())
 	time.Sleep(s.closeListenerDelay)
 	s.listener.Close()
 
-	s.statusLogger.Info("shutting down server in %s", s.osExitDelay.String())
+	s.Logger.Info("shutting down server in %s", s.osExitDelay.String())
 	time.Sleep(s.osExitDelay)
 
 	s.ExitProcess()
 }
 
 func (s *Server) ExitProcess() {
-	s.statusLogger.Info("shutting down server with exit code %d", s.osExitCode)
+	s.Logger.Info("shutting down server with exit code %d", s.osExitCode)
 	os.Exit(s.osExitCode)
 }
 
 // NewMiddlewareHandler wraps the middlewares in a http.Handler. The handler,
 // on activation, calls each middleware in order, if no error was returned and
 // `ctx.Next()` was called. If a middleware wants to finish the processing, it
-// can just write to the `http.ResponseWriter` or use the `ctx.Responder` for
+// can just write to the `http.ResponseWriter` or use the `ctx.Response` for
 // convienience.
-//
-// The `Context.App` can be initialized by providing a CtxConstructor via
-// `SetAppContext()`.
-func (this *Server) NewMiddlewareHandler(middlewares []Middleware) http.Handler {
+func (s *Server) NewMiddlewareHandler(middlewares []Middleware) http.Handler {
 	return http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
-		// Initialize fresh scope variables.
-		ctx := &Context{
-			MuxVars: mux.Vars(req),
-			Response: Response{
-				w: res,
-			},
-		}
-
-		if this.ctxConstructor != nil {
-			ctx.App = this.ctxConstructor()
+		// prepare request
+		reqID := req.Header.Get(RequestHeader)
+		if reqID == "" {
+			reqID = s.Uuid()
 		}
 
-		for _, middleware := range middlewares {
-			nextCalled := false
-			ctx.Next = func() error {
-				nextCalled = true
-				return nil
+		logger := MustGetLogger(LoggerOptions{ID: reqID, Level: s.logLevel})
+
+		// create handler that actually processes the middlewares
+		middlewareHandler := http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+			ctx := &Context{
+				MuxVars:     mux.Vars(req),
+				requestMeta: map[string]interface{}{},
+				Response: Response{
+					w: res,
+				},
+				RequestID: func() string {
```

Comment:
Would be an option, but there would still be the need to stick a the `reqID` to the context to make the getter work. I see no benefit. This just follows the pattern with the `Next()` function, because they are bounded to the request scope.